crates/rpc/rpc/src/debug.rs

```diff
@@ -569,110 +575,80 @@ where
-                    );
-
-                    let storage = hashed_state
-                        .storages
-                        .entry(hashed_address)
-                        .or_insert_with(|| HashedStorage::new(account.status.was_destroyed()));
-
-                    if let Some(account) = account.account {
-                        if include_preimages {
-                            state_preimages
-                                .insert(hashed_address, alloy_rlp::encode(address).into());
-                        }
+                let hashed_state = HashedPostState::from_bundle_state(&bundle_state.state);
+                if include_preimages {
+                    for (address, account) in bundle_state.state {
+                        let hashed_address = keccak256(address);
+                        state_preimages.insert(hashed_address, alloy_rlp::encode(address).into());
 
-                        for (slot, value) in account.storage {
+                        for (slot, _) in account.storage {
                             let slot = B256::from(slot);
                             let hashed_slot = keccak256(slot);
-                            storage.storage.insert(hashed_slot, value);
-
-                            if include_preimages {
-                                state_preimages.insert(hashed_slot, alloy_rlp::encode(slot).into());
-                            }
+                            state_preimages.insert(hashed_slot, alloy_rlp::encode(slot).into());
                         }
                     }
                 }
+                // for (address, account) in bundle_state.state {
+                //     let hashed_address = keccak256(address);
+                //     hashed_state.accounts.insert(
+                //         hashed_address,
+                //         account.account.as_ref().map(|a| a.info.clone().into()),
+                //     );
+
+                //     let storage = hashed_state
+                //         .storages
+                //         .entry(hashed_address)
+                //         .or_insert_with(|| HashedStorage::new(account.status.was_destroyed()));
+
+                //     if let Some(account) = account.account {
+                //         if include_preimages {
+                //             state_preimages
+                //                 .insert(hashed_address, alloy_rlp::encode(address).into());
+                //         }
+
+                //         for (slot, value) in account.storage {
+                //             let slot = B256::from(slot);
+                //             let hashed_slot = keccak256(slot);
+                //             storage.storage.insert(hashed_slot, value);
+
+                //             if include_preimages {
+                //                 state_preimages.insert(hashed_slot,
+                // alloy_rlp::encode(slot).into());             }
+                //         }
+                //     }
+                // }
```

Comment:
probably should remove this before marking pr ready for review, else mark pr as draft